src/components/modals/ModalWithActionButtons.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { Button, ButtonProps } from '../buttons/Button';
+
+import { BaseModal } from './BaseModal';
+
+export interface ActionButton extends ButtonProps {
+  actionCallback?: (...params: any[]) => any;
+  closeModalOnClick?: boolean;
+}
+
+export interface ModalProps {
+  title: string;
+  contentText: string;
+  open: boolean;
+  handleClose: () => void;
+}
+
+interface ModalActionButtonProps {
+  modalProps: ModalProps;
+  actionButtonList: ActionButton[];
+}
+
+export const ModalWithActionButtons = ({
+  modalProps,
+  actionButtonList,
+}: ModalActionButtonProps) => {
+  return (
+    <BaseModal {...modalProps}>
+      {actionButtonList.map((buttonProps: ActionButton) => {
+        const {
+          name,
+          closeModalOnClick = true,
+          actionCallback = () => {
+            return null;
+          },
+          ...otherProps
+        } = buttonProps;
+
+        const onClickFunction = () => {
+          if (closeModalOnClick) {
+            modalProps.handleClose();
+          }
+
+          actionCallback();
+        };
+
+        return (
+          <Button key={name} onClick={onClickFunction} {...otherProps}>
```

Comment:
but here onClick takes in a function with type () => void as well right? what was that (event) => void thing you were talking about earlier

Reply:
This is what I was talking about 

![image](https://user-images.githubusercontent.com/42761684/93029998-6aafcf00-f5d4-11ea-8948-3a9e0188a4ed.png)